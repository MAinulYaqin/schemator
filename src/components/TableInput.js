import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { MdClose } from "react-icons/lib/md";

import "typeface-roboto";

const Container = styled.div`
  width: 100%;
  height: 36px;
  display: flex;
  align-items: center;
  justify-content: stretch;
  padding: 0px 4px;
  box-sizing: border-box;
`;

export const Select = styled.select`
  background: #333;
  color: white;
  font-family: Roboto;
  font-size: 11px;
  outline: none;
  width: 100%;
  height: 24px;
`;

export const Input = styled.input`
  width: 100%;
  height: 24px;
  color: white;
  background: #111;
  border: 0px;
  outline: none;
  font-family: Roboto;
  font-size: 12px;
  box-sizing: border-box;
  padding: 0px 8px;
`;

export const CloseButton = styled(MdClose)`
  color: #444;
  font-size: 16px;
  padding: 8px;
  cursor: pointer;
`;

const Column = styled.div`
  flex: 1;
  padding: 4px;
`;

const TableInput = ({
  name,
  type,
  onClickRemove,
  onChangeName,
  onChangeType
}) => {
  const types = [
    "BINARY",
    "BOOLEAN",
    "CHAR",
    "DATE",
    "DATE_TIME",
    "DATE_TIME_TZ",
    "DECIMAL",
    "DOUBLE",
    "ENUM",
    "FLOAT",
    "INTEGER",
    "IP_ADDRESS",
    "JSON",
    "JSONB",
    "MAC_ADDRESS",
    "MORPHS",
    "NULLABLE_MORPHS",
    "STRING",
    "TEXT",
    "TIME",
    "TIME_TZ",
    "TIMESTAMP",
    "TIMESTAMP_TZ",
    "UUID"
  ];

  return (
    <Container>
      <Column>
        <Input type="text" value={name} onChange={onChangeName} />
      </Column>
      <Column>
        <Select value={type} onChange={onChangeType}>
          {types.map((type, index) => (
            <option key={index} value={type}>
              {type}
            </option>
          ))}
        </Select>
      </Column>
      <CloseButton onClick={onClickRemove} />
    </Container>
  );
};

TableInput.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  onClickRemove: PropTypes.func,
  onChangeName: PropTypes.func,
  onChangeType: PropTypes.func
};

export default TableInput;
