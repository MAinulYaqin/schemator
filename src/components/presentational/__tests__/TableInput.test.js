import React from "react";
import { shallow } from "enzyme";

import TableInput, { CloseButton, Input, Select } from "../TableInput";

describe("TableInput", () => {
  const setup = propOverrides => {
    const props = {
      name: "username",
      type: "string",
      types: [
        {
          id: "string",
          label: "String"
        },
        {
          id: "boolean",
          label: "Boolean"
        },
        {
          id: "char",
          label: "Char"
        }
      ],
      onChangeName: jest.fn(),
      onChangeType: jest.fn(),
      onClickRemove: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<TableInput {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with disabled selection", () => {
    const { wrapper } = setup({ isTypeDisabled: true });

    expect(wrapper.find(Select).props().disabled).toEqual(true);
  });

  it("should calls 'onClickRemove' while Remove button was clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(CloseButton).simulate("click");

    expect(props.onClickRemove).toBeCalled();
  });

  it("should calls 'onChangeName' while field name was changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: "id"
      }
    };

    wrapper.find(Input).simulate("change", event);

    expect(props.onChangeName).toBeCalledWith(event);
  });

  it("should calls 'onChangeType' while field type was changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: "BIGINT"
      }
    };

    wrapper.find(Select).simulate("change", event);

    expect(props.onChangeType).toBeCalledWith(event);
  });
});
