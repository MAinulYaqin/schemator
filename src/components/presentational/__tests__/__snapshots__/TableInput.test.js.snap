// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableInput should renders default 1`] = `
<styled.div>
  <styled.div>
    <styled.input
      onChange={[Function]}
      type="text"
      value="username"
    />
  </styled.div>
  <styled.div>
    <styled.select
      onChange={[Function]}
      value="string"
    >
      <option
        key="string"
        value="string"
      >
        String
      </option>
      <option
        key="boolean"
        value="boolean"
      >
        Boolean
      </option>
      <option
        key="char"
        value="char"
      >
        Char
      </option>
    </styled.select>
  </styled.div>
  <Styled(MdClose)
    onClick={[Function]}
  />
</styled.div>
`;
